not ok yet
failed on make release
```
Executing precompile statements... 1438/1935*** This error is usually fixed by running `make clean`. If the error persists, try `make cleanall`. ***
gmake[1]: Leaving directory '/usr/home/julia/buildbot/w1_builder/package_freebsd64/build'
Killed
gmake[1]: *** [sysimage.mk:89: /usr/home/julia/buildbot/w1_builder/package_freebsd64/build/usr/lib/julia/sys-o.a] Error 1
gmake: *** [Makefile:90: julia-sysimg-release] Error 2
program finished with exit code 2
```

--- a/NEWS.md
+++ b/NEWS.md
@@ -89,6 +89,7 @@ Library changes
 * `eachslice` now works over multiple dimensions; `eachslice`, `eachrow` and `eachcol` return
   a `Slices` object, which allows dispatching to provide more efficient methods ([#32310]).
 * `@kwdef` is now exported and added to the public API ([#46273])
+* An issue with order of operations in `fld1` is now fixed ([#28973])
 
 Standard library changes
 ------------------------

--- a/base/operators.jl
+++ b/base/operators.jl
@@ -842,7 +842,7 @@ julia> x == (fld1(x, y) - 1) * y + mod1(x, y)
 true
 ```
 """
-fld1(x::T, y::T) where {T<:Real} = (m = mod1(x, y); fld(x + y - m, y))
+fld1(x::T, y::T) where {T<:Real} = (m = mod1(x, y); fld((x - m) + y, y))
 function fld1(x::T, y::T) where T<:Integer
     d = div(x, y)
     return d + (!signbit(x ⊻ y) & (d * y != x))

--- a/test/operators.jl
+++ b/test/operators.jl
@@ -274,6 +274,9 @@ end
     end
 
     @test fldmod1(4.0, 3) == fldmod1(4, 3)
+        
+    # issue 28973
+    @test fld1(0.4, 0.9) == fld1(nextfloat(0.4), 0.9) == 1.0
 end
 
 @testset "Fix12" begin

